backend/api/Areas/Research/Controllers/ResearchFileController.cs

```diff
@@ -58,6 +59,106 @@ public IActionResult GetResearchFile(long id)
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{researchFileId:long}/activities")]
+        [HasPermission(Permissions.ResearchFileView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<ActivityInstanceModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "researchfile" })]
+        public IActionResult GetActivitiesForResearchFile(long researchFileId)
+        {
+            //TODO: Fetch actual activity from
+            //var researchFile = _pimsService.ActivityService.GetAllByResearchFileId(researchFileId);
+            List<ActivityInstanceModel> activities = new List<ActivityInstanceModel>();
+            activities.Add(new ActivityInstanceModel()
+            {
+                Id = 1,
+                Description = "GENERAL Activity",
+                ActivityTemplateId = 1,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "GENERAL",
+                    Description = "General Visit ",
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            activities.Add(new ActivityInstanceModel()
+            {
+                Id = 2,
+                Description = "Site Visit Activity",
+                ActivityTemplateId = 2,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "SITEVIS",
+                    Description = "Site Visit ",
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            activities.Add(new ActivityInstanceModel()
+            {
+                Id = 3,
+                Description = "Survey Activity",
+                ActivityTemplateId = 3,
+                ActivityTemplateTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "SURVEY",
+                    Description = "Site Visit ",
+
+                },
+                ActivityStatusTypeCode = new Models.TypeModel<string>()
+                {
+                    Id = "Draft",
+                    Description = "Draft",
+                },
+            });
+            return new JsonResult(activities);
+        }
+
+
+        /// <summary>
+        /// Get the document types.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("activity-templates")]
+        [HasPermission(Permissions.ResearchFileView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<ActivityTemplateModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "activity-templates" })]
+        public IActionResult GetActivityTemplateTypes()
```

Comment:
This looks like a general Template API, IMO it would be better to move it to its own controller since is not necessarely related to research files

Reply:
Since we are keeping all activityinstance related code to research controller as per story developer comments, I think it will be easier to find it here. Let me know if you still want it to be moved.